### PR TITLE
DPS-000: Adding service and controller test

### PR DESCRIPTION
**Pull request by mbashirli**
## Added
- Service test
- Controller test

### CODE REVIEW

File: src/test/groovy/BeerControllerSpec.groovy [hunk 1]
```diff
@@ -0,0 +1,165 @@
...
+    BeerController controller
+    BeerService service
+    MockMvc mockMvc
+    ObjectMapper objectMapper
+    EnhancedRandom random = EnhancedRandomBuilder.aNewEnhancedRandom()
+
+
+    void setup() {
+        objectMapper = new ObjectMapper()
+        objectMapper.registerModule(new JavaTimeModule())
```
Comment (nadir-pb) [inline src/test/groovy/BeerControllerSpec.groovy:32]: why we need it?

File: src/test/groovy/BeerControllerSpec.groovy [hunk 2]
```diff
@@ -0,0 +1,165 @@
...
+        def result = mockMvc.perform(post("/api/v1/beer/addBeer")
+                .contentType(MediaType.APPLICATION_JSON)
+                .content(jsonContent))
+                .andReturn()
+
+
+        then:
+        1 * service.saveNewBeer(beer)
+        result.getResponse().status == 201
+
```
Comment (nadir-pb) [inline src/test/groovy/BeerControllerSpec.groovy:92]: noExceptionThrown() - add it, there is no exception case

File: src/test/groovy/BeerServiceSpec.groovy [hunk 3]
```diff
@@ -0,0 +1,115 @@
...
+    EnhancedRandom random = EnhancedRandomBuilder.aNewEnhancedRandom()
+    ObjectMapper objectMapper
+    BeerServiceImpl beerService
+    Beer testBeer = random.nextObject(Beer)
+    Map<UUID, Beer> testBeerMap = new HashMap<UUID, Beer>()
+
+
+    def setup() {
+        objectMapper = Mock()
+        testBeerMap.put(testBeer.getId(), testBeer)
```
Comment (nadir-pb) [inline src/test/groovy/BeerServiceSpec.groovy:22]: pls do not put it directly in the setup

File: src/test/groovy/BeerServiceSpec.groovy [hunk 4]
```diff
@@ -0,0 +1,115 @@
...
+    }
+
+
+    def "deleteById"() {
+        when:
+        beerService.deleteById(testBeer.getId())
+
+
+        then:
+        testBeerMap.isEmpty()
```
[the remark this answers is not in the document]
Comment (nadir-pb) [reply src/test/groovy/BeerServiceSpec.groovy:88]: add it 1* then which method will call and if it returns smth then return it

File: src/test/groovy/BeerControllerSpec.groovy (hunk 1 above)
Comment (mbashirli) [reply src/test/groovy/BeerControllerSpec.groovy:32]: LocalDateTime is not supported by Jackson by default. This helps resolve the incompatibility issue.

File: src/test/groovy/BeerServiceSpec.groovy (hunk 3 above)
Comment (mbashirli) [reply src/test/groovy/BeerServiceSpec.groovy:22]: Can not add testBeer inside testBeerMap before setup(). Gives NullPointerException.

**Review by nadir-pb — APPROVED** [review]

**Pull request merged**